### ISSUE TITLE
Unable to install with commit sha

### ISSUE THREAD

**Issue by pratarora** [rformassspectrometry/MsCoreUtils#130]
Hi

I am trying to install the package from github to maintain reprocubility, however it is not working for mscorutils. it works for other rformassspec packages. doyou what could be going on? 
it installs it i remove the sha commit.

rformassspectrometry/MetaboCoreUtils@4c181cc74f8e456551ef74debcc25bbb6095d23d, 

others that work fine
rformassspectrometry/MetaboCoreUtils
rformassspectrometry/MetaboAnnotation@4510a6395580d6987dc8fea10791919019c29276, rformassspectrometry/Spectra@1efb9093dfcbdc1b21af0acfe62df924547e347c, rformassspectrometry/CompoundDb@d6beeb12744607b35d1e749062191326332b7e4e

best regards
Prateek

**Comment by lgatto** [comment]
Please install all these package with `BiocManager::install()`.

**Comment by pratarora** [comment]
all the others work with BiocManager::install() except rformassspectrometry/MetaboCoreUtils


BiocManager::install("rformassspectrometry/MetaboAnnotation@4510a6395580d6987dc8fea10791919019c29276")
        }
        if (packageVersion("ProtGenerics") < "1.37.1") {
          remove.packages("ProtGenerics")
          BiocManager::install("rformassspectrometry/ProtGenerics")
        }
        if (packageVersion("Spectra") < "1.15.11") {
          remove.packages("Spectra")
          BiocManager::install("rformassspectrometry/Spectra@1efb9093dfcbdc1b21af0acfe62df924547e347c")
        }
        if (packageVersion("CompoundDb") < "1.9.4") {
          remove.packages("CompoundDb")
          BiocManager::install("rformassspectrometry/CompoundDb@d6beeb12744607b35d1e749062191326332b7e4e")

**Comment by lgatto** [comment]
No, do not set a specific commit, use `BiocManager::install('pkgname')`.

**Comment by pratarora** [comment]
Thanks again for your suggestion. I understand using the package name directly is the standard approach. However, for reproducibility, I need to lock specific versions of these packages. I'm encountering this issue specifically with MsCoreUtils when using a commit SHA. The other packages from rformassspectrometry install fine with the specified commits using BiocManager::install(), but MsCoreUtils doesn't. For example, BiocManager::install("rformassspectrometry/MsCoreUtils@4c181cc") fails, but removing the commit SHA allows the installation. I'd like to understand why this is happening and how to fix it. Relying solely on Bioconductor releases isn't ideal for me because they are linked to specific R versions, and often include updates to other packages that I'm not ready to incorporate into my existing pipelines, as this can cause breakage. Therefore, I need more granular control over package versions. The rformassspectrometry maintainers could perhaps consider creating releases for MsCoreUtils to simplify version control and improve reproducibility for users facing similar constraints.

**Comment by lgatto** [comment]
> However, for reproducibility, I need to lock specific versions of these packages.

If you want to guarantee repeatability (which is what you are aiming if you want the exact same software version), you should look into renv, docker, or archive your libraries. 

> Relying solely on Bioconductor releases isn't ideal for me because they are linked to specific R versions.

But that is how things are supposed to work, and if you don't want to follow these guidelines, you are on your own, and nobody, neither Bioconductor not RforMS teams provide any guarantee. 

> The rformassspectrometry maintainers could perhaps consider creating releases for MsCoreUtils to simplify version control and improve reproducibility for users facing similar constraints.

You are asking us to do extra work, when what you want to achieve is possible by other means. What you want to do, of how you want to do it, is not the right approach for repeatable research.

**Comment by lgatto** [comment]
Also

> often include updates to other packages that I'm not ready to incorporate into my existing pipelines, as this causes breakage.

This is worrisome. In general, you should update (release) packages to get the latest bug fixes. If a pipeline breaks upon a standard release update, it would be essential to understand why this is, as it could point to a more fundamental issue in the pipeline's code.

**Comment by pratarora** [comment]
Dear Prof Gatto,

Thank you for your time. I appreciate your insights and the effort you took to respond. 

> This is worrisome. In general, you should update (release) packages to get the latest bug fixes. 

I understand your emphasis on keeping up-to-date with package releases and have myself enjoyed the seamless integration available through Bioconductor.


>If a pipeline breaks upon a standard release update, it would be essential to understand why this is, as it could point to a more fundamental issue in the pipeline's code.

My approach was driven by specific challenges: a recent package update in Bioconductor introduced significant breaking changes, and updates within the rformassspectrometry ecosystem required navigating between two Bioconductor release cycles. Faced with this, I had to choose between reworking my entire pipeline or locking package versions to maintain stability and reproducibility.

> What you want to do, of how you want to do it, is not the right approach for repeatable research.

My primary goal was to achieve reproducibility and repeatable research. This issue arose from the need to ensure consistent results during an “in-between release cycle” that I was stuck in. While my method maybe atypical, it allowed me to lock versions and prevent unexpected changes, ensuring research consistency.

Notably, using specific commits aligns with Git's strengths for ensuring reproducibility, and Bioconductor's integration with GitHub ideally supports such cases, as I observed for other rformassspectrometry packages.

> You are asking us to do extra work, when what you want to achieve is possible by other means. 

I did not intend to create extra work and acknowledge that my use case might be considered an outlier.


> If you want to guarantee repeatability (which is what you are aiming if you want the exact same software version), you should look into renv, docker, or archive your libraries.

Providing an installation script with specific commits seemed like a straightforward solution for users who may find Docker challenging to navigate. However, I am already exploring the use of Docker images (which is how I am currently navigating my situation) and renv, as you also suggested. 

I thank you once again for your feedback. I value the opportunity to learn and will continue to refine my approach and remain open to learning more effective methods.

Best regards, 
Prateek
